### ISSUE TITLE
`read_all` and `ls_hdf` choke on unsanitized dataset names

### ISSUE THREAD

**Issue by reuster986** [Bears-R-Us/arkouda#971]
Reproducer:

```python
my_arrays = {'foo"0"': ak.arange(100),
             'bar"': ak.arange(100)}
ak.save_all(my_arrays, './bad_dataset_names')
ak.read_all('./bad_dataset_names*')
[JSONDecodeError]

ak.ls_hdf('./bad_dataset_names*')
[JSONDecodeError] 
```

This might seem pretty obscure, but unfortunately we have real datasets whose column names are mangled in this exact way and it is quite difficult to change them upstream. Explicitly supplying the `datasets=` kwarg to leave out the mangled columns doesn't work because `read_all` now calls `get_datasets` to check for existence of all user-specified datasets. Using `allow_errors=True` with `read_all` doesn't seem to help either.

I suspect this behavior was introduced in #956 but I'm not sure. @glitch would you mind taking a look?

**Comment by glitch** [comment]
Yup, I'm on it.

**Comment by glitch** [comment]
Ok, I see what's happening.  In the lshdfMsg changes from #956 / #452 the response message was changed from new lined delimited, free-form text (basically the screen output of `h5ls` command) to a json formatted message.  We manually create the json message due to Chapel 1.24.x bug (fixed in 1.25, I can dig up the ticket if need be) but I didn't escape double-quotes in the dataset name.

Good news is that it should be a pretty easy fix, it doesn't take the server out, and we should be able to eliminate it completely when we remove 1.24 support.  Bad news is that it's a bug :(
